R/middleware.R

```diff
@@ -299,18 +299,37 @@ HandlerManager <- R6Class("HandlerManager",
 
           if (reqSize > maxSize) {
             return(list(status = 413L,
-              headers = list(
-                'Content-Type' = 'text/plain'
-              ),
+              headers = list('Content-Type' = 'text/plain'),
               body = 'Maximum upload size exceeded'))
           }
           else {
             return(NULL)
           }
         },
         call = .httpServer(
-          function (req) {
-            withLogErrors(handlers$invoke(req))
+          function(req,
+                   full = getOption("shiny.fullstacktrace", FALSE),
+                   offset = getOption("shiny.stacktraceoffset", TRUE)) {
+
+             withCallingHandlers(captureStackTraces(handlers$invoke(req)),
+               error = function(cond) {
+                 # Don't print shiny.silent.error (i.e. validation errors)
+                 if (inherits(cond, "shiny.silent.error")) return()
+                 if (isTRUE(getOption("show.error.messages"))) {
+                   printError(cond, full = full, offset = offset)
+                   sanitizeErrors = getOption('shiny.sanitize.errors', TRUE)
+
+                   if (inherits(cond, 'shiny.custom.error') || !sanitizeErrors) {
+                     stop(cond$message, call. = FALSE)
```

Comment:
We've all talked about this already, so I'm just noting it here for future reference: it's necessary to have the `stop()` there (i.e. rethrow the error) because we want to set `.call = FALSE` and we can't do that if we just let the error continue being thrown.